src/common/asb/Asb.c

```diff
@@ -629,6 +629,222 @@ static char* g_desiredEnsureDefaultDenyFirewallPolicyIsSet = NULL;
+    { "Ensure no legacy + entries exist in /etc/passwd", "cc627f0c-3b72-6cc9-36b0-d2a3957431f4", "EnsureNoLegacyPlusEntriesInEtcPasswd" },
+    { "Ensure no legacy + entries exist in /etc/shadow", "1a102d2e-76db-5ccf-3580-ccda064e8df3", "EnsureNoLegacyPlusEntriesInEtcShadow" },
+    { "Ensure no legacy + entries exist in /etc/group", "b487e075-15b4-0df1-550f-d8d5edd78eb4", "EnsureNoLegacyPlusEntriesInEtcGroup" },
+    { "Ensure password expiration is configured", "6ee7250d-142f-57db-25d0-c58207135059", "EnsurePasswordExpiration" },
+    { "Ensure password expiration warning days is configured", "b7ac978f-0963-a61f-58ae-5cf84ab76b5d", "EnsurePasswordExpirationWarning" },
+    { "Ensure password reuse is limited", "57bdeae2-863e-14fb-a463-d6ee64816b33", "EnsurePasswordReuseIsLimited" },
+    { "Ensure the password hashing algorithm is SHA-512", "01ec5346-882b-485d-8960-01dedd608792", "EnsurePasswordHashingAlgorithm" },
+    { "Ensure the minimum days between password changes is configured", "50272a84-672d-4c11-a67e-9d058adaf67a", "EnsureMinDaysBetweenPasswordChanges" },
+    { "Ensure the maximum days between password changes is configured", "7F84F3A4-360D-494C-AA1E-1E15AE9FC9AC", "EnsureMaxDaysBetweenPasswordChanges" },
+    { "Ensure the inactive password lock period is configured", "91fbaeac-f5d0-4ac9-aa1b-52215aef1ed8", "EnsureInactivePasswordLockPeriod" },
+    { "Ensure all system accounts are non-login", "448b668a-738c-420b-b332-51ea49922933", "EnsureSystemAccountsAreNonLogin" },
+    { "Ensure default group for the root account is GID 0", "732fa92f-647e-47b8-b5a4-fdf00b02d9e2", "EnsureDefaultRootAccountGroupIsGidZero" },
+    { "Ensure root is the only UID 0 account", "b5845ff3-42f4-4112-b2a2-5b827232a053", "EnsureRootIsOnlyUidZeroAccount" },
+    { "Ensure unnecessary accounts are removed", "627b7494-0e62-4093-9f77-db8d526d036b", "EnsureUnnecessaryAccountsAreRemoved" },
+    { "Ensure the auditd package is installed", "f9fd03d2-75e4-4564-84a9-4e955f1e7c30", "EnsureAuditdInstalled" },
+    { "Ensure the auditd service is running", "c146c4a4-5eb6-4205-88da-5a71a82f2d45", "EnsureAuditdServiceIsRunning" },
+    { "Ensure the SNMP Server is not enabled", "ca1aea32-3969-49ab-abfc-2c5796a9a8bb", "EnsureSnmpServerIsDisabled" },
+    { "Ensure the rsync service is not enabled", "63aba603-b1f8-40df-82c5-38915452ce23", "EnsureRsynServiceIsDisabled" },
+    { "Ensure the NIS server is not enabled", "b4ad3fdd-7b68-4b11-a3ed-84b37a68b995", "EnsureNisServerIsDisabled" },
+    { "Ensure the rsh client is not installed", "6d441f31-f888-4f4f-b1da-7cfc26263e3f", "EnsureRshClientNotInstalled" },
+    { "Ensure SMB V1 with Samba is disabled", "7624efb0-3026-4c72-8920-48d5be78a50e", "EnsureSmbWithSambaIsDisabled" }
+};
+
+int AsbIsValidResourceIdRuleId(const char* resourceId, const char* ruleId, const char* payloadKey, void* log)
```

Comment:
Is this meant to reject any `RuleId` which is not known? Is there a concern with hard-coding that?

Reply:
Any rule id and rule name that does not match the expected values for that rule. We need the reference, and the choice here is to put this into the code so the NRP can also use it at run time. The audits will still work, but the reason codes will  signal the problems. Please reactivate this if you still have questions.